cmd/veneur/main.go

```diff
@@ -94,38 +100,60 @@ func main() {
 
 	packetPool := &sync.Pool{
 		New: func() interface{} {
-			return make([]byte, veneur.Config.MetricMaxLength)
+			return packet{
+				buf: make([]byte, veneur.Config.MetricMaxLength),
+				ip:  "",
+			}
 		},
 	}
 
+	veneurIPdigester := fnv.New32()
+	veneurIPdigester.Write([]byte("veneur.unique_sender_ips"))
+	veneurIPdigester.Write([]byte("set"))
+	uniqueSenderIPDigest := veneurIPdigester.Sum32()
+
 	// Creates N workers to handle incoming packets, parsing them,
 	// hashing them and dispatching them on to workers that do the storage.
-	parserChan := make(chan []byte)
+	parserChan := make(chan packet)
 	for i := 0; i < veneur.Config.NumWorkers; i++ {
 		go func() {
 			defer func() {
 				veneur.ConsumePanic(recover())
 			}()
-			for packet := range parserChan {
-				handlePacket(workers, packet)
+			for packetbuf := range parserChan {
+				handlePacket(workers, packetbuf.buf)
+				// we don't want to talk to statsd in the packet-processing code
+				// path - it causes packet drops (even in the parsing loop).
+				// instead, we can push the metric to a worker and let it handle
+				// the IO at flush time, which has much less performance impact
+				workers[uniqueSenderIPDigest%uint32(veneur.Config.NumWorkers)].WorkChan <- veneur.Metric{
```

Comment:
This makes for a hot worker. That might be ok? Something to consider. Maybe add a warning about that here?